extension-style-app/src/androidTest/java/com/mapbox/maps/testapp/style/sources/generated/GeoJsonSourceTest.kt

```diff
@@ -75,24 +75,64 @@ class GeoJsonSourceTest : BaseStyleTest() {
   }
 
   @Test
-  @UiThreadTest
   fun urlTest() {
+    val latch = CountDownLatch(2)
+    val answerList = mutableListOf<String?>()
     val testSource = geoJsonSource(SOURCE_ID) {
       url(TEST_URI)
     }
-    setupSource(testSource)
-    assertEquals(TEST_URI, testSource.data)
+    val listener = OnSourceDataLoadedListener {
+      if (it.type == SourceDataType.METADATA && it.id == SOURCE_ID) {
+        answerList.add(testSource.data)
+        latch.countDown()
+      }
+    }
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        mapboxMap.apply {
+          addOnSourceDataLoadedListener(listener)
+          setupSource(testSource)
+        }
+      }
+    }
+    if (!latch.await(LATCH_MAX_TIME_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+    assertEquals(2, answerList.size)
+    assertEquals("", answerList[0])
+    assertEquals(TEST_URI, answerList[1])
+    mapboxMap.removeOnSourceDataLoadedListener(listener)
```

Comment:
Nit: should we always remove the listener? Also, used the return of `await` to assert instead of throwing?
```suggestion
    try {
      assertTrue(latch.await(LATCH_MAX_TIME_MS, TimeUnit.MILLISECONDS))
      assertEquals(2, dataList.size)
      assertEquals("", dataList[0])
      assertEquals(TEST_URI, dataList[1])
    } finally {
      mapboxMap.removeOnSourceDataLoadedListener(listener)
    }
```